QuestPDF.Previewer/PreviewerWindowViewModel.cs

```diff
@@ -0,0 +1,98 @@
+﻿using QuestPDF.Fluent;
+using System.Diagnostics;
+using ReactiveUI;
+using QuestPDF.Infrastructure;
+using Unit = System.Reactive.Unit;
+using Avalonia.Threading;
+
+namespace QuestPDF.Previewer
+{
+    internal class PreviewerWindowViewModel : ReactiveObject
+    {
+        public DocumentRenderer DocumentRenderer { get; } = new();
+
+        private IDocument? _document;
+        public IDocument? Document
+        {
+            get => _document;
+            set
+            {
+                this.RaiseAndSetIfChanged(ref _document, value);
+                UpdateDocument(value);
+            }
+        }
+
+        private float _currentScroll;
+        public float CurrentScroll
+        {
+            get => _currentScroll;
+            set => this.RaiseAndSetIfChanged(ref _currentScroll, value);
+        }
+
+        private float _scrollViewportSize;
+        public float ScrollViewportSize
+        {
+            get => _scrollViewportSize;
+            set
+            {
+                this.RaiseAndSetIfChanged(ref _scrollViewportSize, value);
+                VerticalScrollbarVisible = value < 1;
+            }
+        }
+
+        private bool _verticalScrollbarVisible;
+        public bool VerticalScrollbarVisible
+        {
+            get => _verticalScrollbarVisible;
+            private set => Dispatcher.UIThread.Post(() => this.RaiseAndSetIfChanged(ref _verticalScrollbarVisible, value));
+        }
+
+        public ReactiveCommand<Unit, Unit> ShowPdfCommand { get; }
+
+        public PreviewerWindowViewModel()
+        {
+            HotReloadManager.UpdateApplicationRequested += InvalidateDocument;
+            ShowPdfCommand = ReactiveCommand.Create(ShowPdf);
+        }
+
+        public void UnregisterHotReloadHandler()
+        {
+            HotReloadManager.UpdateApplicationRequested -= InvalidateDocument;
+        }
+
+        private void InvalidateDocument(object? sender, EventArgs e)
+        {
+            UpdateDocument(Document);
+        }
+
+        private Task UpdateDocument(IDocument? document)
+        {
+            return Task.Run(() => DocumentRenderer.UpdateDocument(document));
+        }
+
+        private void ShowPdf()
+        {
+            var path = Path.Combine(Path.GetTempPath(), ".pdf");
```

Comment:
I copied this from your implementation in the PreviewerWindow.
However the path seems weird to me, shouldnt we give the file an actual name?
Right now the file is just called '.pdf'.

Reply:
You are right! Great catch. Not sure why it was working just fine but definitelly we want to assign a filename. Will replace with `Path.GetTempFileName()` call 😁